linux/kernel/building.md

```diff
@@ -1,8 +1,8 @@
 # Kernel building
 
-There are two main methods for building the kernel. You can build locally on a Raspberry Pi, which will take a long time; or you can cross-compile, which is much quicker, but requires more setup.
+There are two ways of compiling the linux kernel. One is to compile nativly or cross architecture via cross compiling. Cross compiling is used to compile the kernel for an architecture (such as arm64) on when on another architecture (such as armhf on the Raspberry Pi OS 32-bit).
```

Comment:
I meant this paragraph probably needs updating but I added some starting information

Reply:
Something like this might be better?

There are two main methods for building the kernel - natively and cross-compiling. Native compilation is the easiest approach, and is used when the architecture you're building the kernel on (host platform) is the same as the architecture you'll be running it on (target platform); such as building a 32-bit Raspberry Pi kernel using 32-bit Raspberry Pi OS, or building a 64-bit Raspberry Pi kernel using 64-bit Raspberry Pi OS. Cross-compiling is used when the "host platform" is different to the "target platform", such as building a 64-bit Raspberry Pi kernel using 32-bit Raspberry Pi OS (or vice versa), or when building a Raspberry Pi kernel (either 32 or 64-bit) on an separate Linux PC. Cross-compilation is slightly more involved, but building a Raspberry Pi kernel is typically much faster when cross-compiled on a separate Linux PC than when built natively on a Raspberry Pi.

I suspect @pelwell and @JamesH65 may have other suggestions / tweaks to the above too. Not sure if what I've suggested here is too wordy?

Reply:
My attempt:
> The default compilers and linkers that come with an OS are configured to build executables to run on that OS - they are _native_ tools - but that doesn't have to be the case. A cross-compiler is configured to build code for a target other than the one running the build process, and using it is called cross-compilation.
>
> Cross-compilation of the Raspberry Pi kernel is useful for two reasons:
> 1. it allows a 64-bit kernel to be built using a 32-bit OS, and vice versa, and
> 2. even a modest laptop can cross-compile a Pi kernel significantly faster than the Pi itself.
>
> The instructions below are divided into native builds and cross-compilation; choose the section appropriate for your situation - although there are many common steps between the two there are also some important differences.

Reply:
I personally like @pelwell 's attempt as it is not as wordy. Perhaps we should add hyper links to the section of the document in the header?

Reply:
I will add pelwell's header for now